e2e/tests/test_snapp_deposit_withdraw.rs

```diff
@@ -0,0 +1,152 @@
+use e2e::common::{wait_for, wait_for_condition, FutureWaitExt};
+use e2e::snapp::{await_state_transition, setup_snapp};
+use ethcontract::web3::api::Web3;
+use ethcontract::web3::transports::Http;
+use ethcontract::web3::types::{H160, H256, U128, U256};
+use ethcontract::Account;
+use std::str::FromStr;
+
+#[test]
+fn test_deposit_and_withdraw() {
+    let (eloop, http) = Http::new("http://localhost:8545").expect("transport failed");
+    eloop.into_remote();
+    let web3 = Web3::new(http);
+    let (instance, accounts, tokens, db) = setup_snapp(&web3, 3, 3);
+
+    // Test environment values
+    let deposit_amount = 18_000_000_000_000_000_000u128;
+
+    let user_address = &accounts[2];
+    let user_id = instance
+        .public_key_to_account_map(*user_address)
+        .call()
+        .wait()
+        .expect("Could not recover account id");
+    let db_account_id = H160::from_low_u64_be(user_id);
```

Comment:
This part is little confusing, it seams that we are using addresses as account indices for Snapp?

**edit**: After talking offline to @bh2smith it looks like this is because we are sharing `AccountState` between StableX and Snapp and they keep track of accounts differently, and this was the interim solution.

**edit 2**: There is also a TODO futher down in the code.

Reply:
I moved the TODO up to here.